### PR TITLE
#35: Course tiles getting indicators for different states

### LINKED ISSUE #35: Highlight different course states with colors
### Details
Depending on the status of a course the background color of the text field in a course tile is changed to a status specific color.

Desktop: https://www.figma.com/file/TtucXsvBOfZE7NZKzTimrA/edu-Wireframe-–-WIP?node-id=998%3A0

### PR DESCRIPTION
**Pull request by okwast**
Depends on #48 

closes #35

### CODE REVIEW

File: frontend/queries/courseFragment.ts [hunk 1]
```diff
@@ -17,7 +17,15 @@ export const COURSE_FRAGMENT = gql`
     MaxParticipants
     Name
     OnlineCourses
-    SemesterId
+    Semester {
```
Comment (fschindler) [inline frontend/queries/courseFragment.ts]: For both roles (`user` and `anonymous`) there is an error:
```
{"errors":[{"extensions":{"path":"$.selectionSet.Course.selectionSet.Semester","code":"validation-failed"},"message":"field \"Semester\" not found in type: 'Course'"}]}
```

Both roles doesn't have any select permission on the `Semester` table.

File: frontend/components/course/Tile.tsx [hunk 2]
```diff
@@ -2,17 +2,84 @@ import Image from "next/image";
...
+
 export const Tile: FC<IProps> = ({ course }) => {
+  const enrollmentStatus = enrollmentStatusForCourse(course);
+
+  const semester = course.Semester;
+
+  const currentDate = new Date();
+  const isCurrentSemester =
+    !!semester &&
+    semester?.Start <= currentDate &&
```
Comment (fschindler) [inline frontend/components/course/Tile.tsx]: According to https://github.com/edu-hub-project/edu-hub/issues/35#issuecomment-882622783 this should be:
```suggestion
    semester?.ApplicationStart <= currentDate &&
```

File: frontend/components/course/Tile.tsx [hunk 3]
```diff
@@ -2,17 +2,84 @@ import Image from "next/image";
...
+    currentDate <= semester?.End;
```
Comment (fschindler) [inline frontend/components/course/Tile.tsx]: According to https://github.com/edu-hub-project/edu-hub/issues/35#issuecomment-882622783 this should be:
```suggestion
    currentDate <= semester?.PerformanceRecordDeadline;
```

File: frontend/components/course/Tile.tsx [hunk 4]
```diff
@@ -2,17 +2,84 @@ import Image from "next/image";
...
+  if (
+    status === EnrollmentStatus_enum.REJECTED ||
+    status === EnrollmentStatus_enum.ABORTED
+  ) {
+    return "bg-gray-300";
+  }
+  if (status === EnrollmentStatus_enum.INVITED) {
+    return "bg-edu-course-invited";
+  }
+  return "";
```
Comment (fschindler) [inline frontend/components/course/Tile.tsx:46]: As you already mentioned: The `EXPIRED` status is  missing.

**Review by fschindler — CHANGES_REQUESTED** [review]
Great changes!

We need to fix the permissions in Hasura. Feel free to give some feedback to my other comments.

File: frontend/queries/courseFragment.ts (hunk 1 above)
Comment (okwast) [reply frontend/queries/courseFragment.ts]: I fixed the permissions and after the new hasura migrations fixed them again. I hope they are there now.

File: frontend/components/course/Tile.tsx (hunk 2 above)
Comment (okwast) [reply frontend/components/course/Tile.tsx]: Good catch. Fixed that.

File: frontend/components/course/Tile.tsx (hunk 3 above)
Comment (okwast) [reply frontend/components/course/Tile.tsx]: Also fixed this.

File: frontend/components/course/Tile.tsx (hunk 4 above)
Comment (okwast) [reply frontend/components/course/Tile.tsx:46]: Added it now that I have the data.

File: frontend/components/course/Tile.tsx [hunk 5]
```diff
@@ -2,17 +2,99 @@ import Image from "next/image";
...
+    return "bg-edu-course-current";
+  }
+  if (
+    status === EnrollmentStatus_enum.REJECTED ||
+    status === EnrollmentStatus_enum.ABORTED
+  ) {
+    return "bg-gray-300";
+  }
+  if (status === EnrollmentStatus_enum.INVITED) {
+    if (enrollment && new Date() < enrollment.ExpirationDate) {
```
Comment (fschindler) [inline frontend/components/course/Tile.tsx]: `new Date() < null === false` -> enrollments without an expiration date are always expired?

File: frontend/components/course/Tile.tsx [hunk 6]
```diff
@@ -2,17 +2,99 @@ import Image from "next/image";
...
+      return "bg-edu-course-invited";
+    } else {
+      return "bg-gray-300";
```
Comment (fschindler) [inline frontend/components/course/Tile.tsx:43]: Having a comment here could be helpful for the future.
```suggestion
     // invitation is expired
     return "bg-gray-300";
```

**Review by fschindler — CHANGES_REQUESTED** [review]

File: frontend/components/course/Tile.tsx (hunk 5 above)
Comment (okwast) [reply frontend/components/course/Tile.tsx]: I changed it so if there is no expiration date the invitation is valid.

**Review by fschindler — APPROVED** [review]
This branch is now up to date with `develop` and I'm going to merge it.

**Pull request merged**